app/models/miq_report.rb

```diff
@@ -53,14 +53,22 @@ class MiqReport < ActiveRecord::Base
   # Scope on reports that have report results.
   #
   # Valid options are:
-  #   ::miq_group:    An MiqGroup instance on which to filter
-  #   ::miq_group_id: An MiqGroup id on which to filter
+  #   ::miq_groups:    An MiqGroups instance on which to filter
+  #   ::miq_group_ids: An MiqGroup ids on which to filter
   #   ::select:       An Array of MiqReport columns to fetch
   def self.having_report_results(options = {})
-    q = joins(:miq_report_results).order(arel_table[:id])
+    miq_group_ids = options[:miq_groups].nil? ? nil : options[:miq_groups].collect(&:id)
+    miq_group_ids ||= options[:miq_group_ids]
 
-    miq_group = MiqGroup.extract_ids(options[:miq_group] || options[:miq_group_id])
-    q = q.where(MiqReportResult.arel_table[:miq_group_id].eq(miq_group)) unless miq_group.nil?
+    miq_group_condition = {:miq_report_results => {:miq_group_id => miq_group_ids}}
+
+    if miq_group_ids.nil?
+      miq_report_with_condition = where.not(miq_group_condition)
+    else
+      miq_report_with_condition = where(miq_group_condition)
+    end
```

Comment:
@lpichler Can you try moving this `where` construction to a method like "group_filter_from_options" or something named better that returns the `.where` that you can chain below?  

It feels like this method is doing too much.